app/frontend/src/features/profile/hooks/useUpdateHostingPreferences.test.tsx

```diff
@@ -3,39 +3,64 @@ import useUpdateHostingPreferences from "features/profile/hooks/useUpdateHosting
 import useCurrentUser from "features/userQueries/useCurrentUser";
 import { Empty } from "google-protobuf/google/protobuf/empty_pb";
 import { act } from "react-test-renderer";
-import { service } from "service";
+import { HostingPreferenceData, service } from "service";
 import wrapper from "test/hookWrapper";
 import { addDefaultUser } from "test/utils";
+import { Primitives, ServicePrimitives } from "utils/types";
 
 const getUserMock = service.user.getUser as jest.Mock;
 const updateHostingPreferenceMock = service.user
   .updateHostingPreference as jest.Mock;
 
 describe("useUpdateHostingPreference hook", () => {
-  const newHostingPreferenceData = {
-    acceptsKids: false,
+  const newHostingPreferenceData: HostingPreferenceData = {
     acceptsPets: false,
     area: "",
     houseRules: "",
     lastMinute: true,
     maxGuests: null,
     smokingAllowed: 1,
     wheelchairAccessible: true,
+    hasPets: false,
+    petDetails: "",
+    hasKids: false,
+    acceptsKids: false,
+    kidDetails: "",
+    hasHousemates: false,
+    housemateDetails: "",
+    smokesAtHome: false,
+    drinkingAllowed: false,
+    drinksAtHome: false,
+    otherHostInfo: "",
+    campingOk: true,
+    parking: true,
+    parkingDetails: 3,
+    sleepingArrangement: 2,
+    sleepingDetails: "",
   };
 
   it("updates the store with the latest user hosting preference", async () => {
     addDefaultUser();
-    const newUserPref = Object.entries(newHostingPreferenceData).reduce(
-      (acc: Record<string, unknown>, [key, value]) => {
-        if (key !== "smokingAllowed") {
-          acc[key] = { value };
-        } else {
+
+    const newUserPref = (Object.entries(newHostingPreferenceData) as [
+      keyof HostingPreferenceData,
+      Primitives
+    ][]).reduce((acc, [key, value]) => {
```

Comment:
```suggestion
    ][]).reduce((acc: Record<string, ServicePrimitives>, [key, value]) => {
```
Just personal preference to avoid type assertions if things work without it for reasons mentioned in the other comment